packages/dicomImageLoader/src/imageLoader/wadors/getPixelData.ts

```diff
@@ -1,100 +1,66 @@
 
-function uint8ArrayToString(data, offset, length) {
-  offset = offset || 0;
-  length = length || data.length - offset;
-  let str = '';
-
-  for (let i = offset; i < offset + length; i++) {
-    str += String.fromCharCode(data[i]);
-  }
+import { xhrRequest } from '../internal/index';
+// import rangeRequest from '../internal/rangeRequest';
+import streamRequest from '../internal/streamRequest';
+import rangeRequest from '../internal/rangeRequest';
+import extractMultipart from './extractMultipart';
+import { LossyConfiguration } from 'core/src/types';
 
-  return str;
-}
+const { ProgressiveIterator } = utilities;
 
 function getPixelData(
   uri: string,
   imageId: string,
-  mediaType = 'application/octet-stream'
-): Promise<any> {
+  mediaType = 'application/octet-stream',
+  retrieveOptions: LossyConfiguration = {}
+) {
   const headers = {
     Accept: mediaType,
   };
 
-  return new Promise((resolve, reject) => {
-    const loadPromise = xhrRequest(uri, imageId, headers);
-    const { xhr } = loadPromise;
-
-    loadPromise.then(function (imageFrameAsArrayBuffer /* , xhr*/) {
-      // request succeeded, Parse the multi-part mime response
-      const response = new Uint8Array(imageFrameAsArrayBuffer);
-
-      const contentType =
-        xhr.getResponseHeader('Content-Type') || 'application/octet-stream';
-
-      if (contentType.indexOf('multipart') === -1) {
-        resolve({
-          contentType,
-          imageFrame: {
-            pixelData: response,
-          },
-        });
-
-        return;
-      }
-
-      // First look for the multipart mime header
-      const tokenIndex = findIndexOfString(response, '\r\n\r\n');
-
-      if (tokenIndex === -1) {
-        reject(new Error('invalid response - no multipart mime header'));
-      }
-      const header = uint8ArrayToString(response, 0, tokenIndex);
-      // Now find the boundary  marker
-      const split = header.split('\r\n');
-      const boundary = findBoundary(split);
-
-      if (!boundary) {
-        reject(new Error('invalid response - no boundary marker'));
-      }
-      const offset = tokenIndex + 4; // skip over the \r\n\r\n
+  // TODO - consider allowing a complete rewrite of the path
+  let url = retrieveOptions?.urlArguments
+    ? `${uri}${uri.indexOf('?') === -1 ? '?' : '&'}${
+        retrieveOptions.urlArguments
+      }`
+    : uri;
+  if (retrieveOptions?.framesPath) {
+    url = url.replace('/frames/', retrieveOptions.framesPath);
+  }
 
-      // find the terminal boundary marker
-      const endIndex = findIndexOfString(response, boundary, offset);
+  if (retrieveOptions.byteRange) {
+    return rangeRequest(url, imageId, headers, retrieveOptions);
+  }
 
-      if (endIndex === -1) {
-        reject(new Error('invalid response - terminating boundary not found'));
-      }
+  // Default to streaming the response data so that it can be decoding in
+  // a streaming parser.
+  if (retrieveOptions.streaming !== false) {
```

Comment:
why not === true?

Reply:
Because I want undefined to default to true here so that one can try streaming by default, and then stop streaming if it isn't supported after having read the transfer syntax.